### PR TITLE
refactor(my collection): reuse auction result component

### PR DESCRIPTION
**Pull request by admbtlr**
The type of this PR is: Refactor

### Description

Extracts `AuctionResult` (and `AuctionResultFragmentContainer`) from `ArtistInsightAuctionResults`, moves them into a directory with other list components, and reuses them in `MyCollectionAuctionResults`.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

### CODE REVIEW

File: src/lib/Components/Lists/AuctionResult.tsx [hunk 1]
```diff
@@ -94,9 +94,15 @@ const ArtistInsightsAuctionResult: React.FC<Props> = ({ auctionResult }) => {
                 )}
               </Flex>
             ) : (
-              <Text variant="mediumText" style={{ width: 70 }} textAlign="right">
-                {isFromPastMonth ? "Awaiting results" : auctionResult.boughtIn === true ? "Bought in" : "Not available"}
-              </Text>
+              <Flex alignItems="flex-end">
+                <Text variant="subtitle" fontWeight="bold" style={{ width: 70 }} textAlign="right">
+                  {isFromPastMonth
```
Comment (brainbicycle) [inline src/lib/Components/Lists/AuctionResult.tsx:99]: I think @pvinis recently updated this logic in this PR https://github.com/artsy/eigen/pull/4281, hopefully git will figure it out but just something to keep an eye on to make sure we get updated logic on merge!

File: src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/MyCollectionArtworkArtistAuctionResults.tsx [hunk 2]
```diff
@@ -125,21 +95,8 @@ export const MyCollectionArtworkArtistAuctionResultsFragmentContainer = createFr
...
-                }
-                description
-                dateText
-                saleDate
-                priceRealized {
-                  display
-                  centsUSD
-                }
+                id
+                ...AuctionResult_auctionResult
```
Comment (brainbicycle) [inline src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/MyCollectionArtworkArtistAuctionResults.tsx:99]: ✨

**Review by brainbicycle — APPROVED** [review]
Looks good to me 👍, thanks for pushing for consistency here. 
Is it possible to add a screenshot with the updated component? 
Anna was also working on a ticket in this same area not sure if the existing component has a placeholder image but if not might be a good one to pair on:
https://artsyproduct.atlassian.net/browse/CX-913

**Comment by admbtlr** [comment]
> Is it possible to add a screenshot with the updated component?

Done - it's pretty much the same, although I tweaked the whitespace slightly.

**Comment by brainbicycle** [comment]
Is this ready to be merged? I didn't merge because I wasn't assigned but looks good to me!

**Pull request merged**